carina-api/src/main/java/com/qaprosoft/carina/core/foundation/api/AbstractApiMethodV2.java

```diff
@@ -236,14 +238,21 @@ public void validateResponse(String... validationFlags) {
         validateResponse(JSONCompareMode.NON_EXTENSIBLE, validationFlags);
     }
 
-    public void validateResponseAgainstJSONSchema(String schemaPath) {
+    public void validateResponseAgainstSchema(String schemaPath) {
         if (actualRsBody == null) {
             throw new RuntimeException("Actual response body is null. Please make API call before validation response");
         }
-        TemplateMessage tm = new TemplateMessage();
-        tm.setTemplatePath(schemaPath);
-        String schema = tm.getMessageText();
-        JsonValidator.validateJsonAgainstSchema(schema, actualRsBody);
+        ContentType contentType = this.getClass().getAnnotation(ContentType.class);
+        if (contentType == null || contentType.type().equals(JSON_CONTENT_TYPE.getHeaderValue())) {
+            TemplateMessage tm = new TemplateMessage();
+            tm.setTemplatePath(schemaPath);
+            String schema = tm.getMessageText();
+            JsonValidator.validateJsonAgainstSchema(schema, actualRsBody);
+        } else if (contentType.type().equals(XML_CONTENT_TYPE.getHeaderValue())) {
+            XmlValidator.validateXmlAgainstSchema(schemaPath, actualRsBody);
+        } else {
+            throw new RuntimeException("Unsupported argument of content type");
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Define and throw a dedicated exception instead of using a generic one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://ci.qaprosoft.com/sonarqube/coding_rules#rule_key=squid%3AS00112)